File: src/test/java/pirateboat/torrent/PremiumizeTest.java
package pirateboat.torrent;

import pirateboat.info.TheFilmDataBaseService;
import pirateboat.multifileHoster.Premiumize;
import pirateboat.utilities.HttpHelper;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class PremiumizeTest {

    Premiumize premiumize;

    @BeforeEach
    void beforeEach() {
        premiumize = new Premiumize(new HttpHelper());
    }

    @Test
    void getCacheStateOfTorrent() {
        // Given
        List<Torrent> listOfTorrents = new ArrayList<>();
        Torrent torrentToBeChecked = new Torrent("Test");
        torrentToBeChecked.magnetUri="magnet:?xt=urn:btih:e2467cbf021192c241367b892230dc1e05c0580e&dn=test";
        listOfTorrents.add(torrentToBeChecked);

        // When
        premiumize.enrichCacheStateOfTorrents(listOfTorrents);
        // Then
        assertTrue(listOfTorrents.get(0).cached.size()>0);
    }

    @Test
    void getCacheStateOfTorrents() {
        // Given
        List<Torrent> listOfTorrents = new ArrayList<>();
        Torrent torrentToBeChecked = new Torrent("Test");
        torrentToBeChecked.magnetUri="magnet:?xt=urn:btih:e2467cbf021192c241367b892230dc1e05c0580e&dn=test";
        listOfTorrents.add(torrentToBeChecked);
        listOfTorrents.add(torrentToBeChecked);

        // When
        premiumize.enrichCacheStateOfTorrents(listOfTorrents);
        // Then
        assertTrue(listOfTorrents.get(0).cached.size()>0);
    }

    @Test()
    void getCacheStateOfTorrentsWithEMptyDN() {
        // Given
        List<Torrent> listOfTorrents = new ArrayList<>();
        Torrent torrentToBeChecked = new Torrent("Test");
        torrentToBeChecked.magnetUri="magnet:?xt=urn:btih:e2467cbf021192c241367b892230dc1e05c0580e&tracker=test";
        listOfTorrents.add(torrentToBeChecked);

        // When
        premiumize.enrichCacheStateOfTorrents(listOfTorrents);
        // Then
        assertTrue(listOfTorrents.get(0).cached.size()>0);
    }
}

File: src/test/java/boat/multifileHoster/PremiumizeTest.kt
package boat.multifileHoster

import boat.model.TransferStatus
import boat.torrent.Torrent
import boat.utilities.HttpHelper
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertDoesNotThrow

class PremiumizeTest {

    private val premiumize: Premiumize = Premiumize(HttpHelper())

    @Test
    fun getRemainingTrafficInMB() {
        // Given
        // When
        // Then
        assertDoesNotThrow { premiumize.getRemainingTrafficInMB() }
    }

    @Test
    fun getRemoteTorrents() {
        // Given
        // When
        // Then
        assertDoesNotThrow { premiumize.getRemoteTorrents() }
    }

    @Test
    fun filesFromTorrent() {
        // Given
        val remoteTorrents: List<Torrent?> = premiumize.getRemoteTorrents()
        // When
        val torrent = remoteTorrents.firstOrNull { torrent -> torrent?.remoteTransferStatus?.equals(TransferStatus.READY_TO_BE_DOWNLOADED) ?: false }
        if (torrent != null) {
            val filesFromTorrent = premiumize.getFilesFromTorrent(torrent)
            Assertions.assertNotNull(filesFromTorrent)
        }
        Assertions.assertTrue(true)
    }
}